### DISCUSSION TITLE
One to One relationship "afterCreate" lifecycle method not called on related resources

### DISCUSSION THREAD

**Discussion by danutavadanei** [laravel/nova-issues#5379] [Ideas & Feature Requests]
- Laravel Version: 8.2
- Nova Version: v4.21.0 (Silver Surfer)
- PHP Version: 8.2
- Database Driver & mysql 8.0
- Operating System and Version: linux

### Description:

If you create a resource that has a `HasOne` or a `MorphOne` related resource, the related resource `afterCreate` method is not called.

**Comment by crynobone** [comment]
Moved this into "Idea & Feature Requests". We need to first decide whether `afterCreate` and `afterUpdate` should be triggered when a resource is created/updated inline.

**Reply by danutavadanei** [reply]
From a developer perspective I expect the behavior to be the same on a resource if I do actions directly on it or via a related resource if not mentioned otherwise.

**Reply by puzzledmonkey** [reply]
seconded - from a developer perspective there should be no difference in the lifecycle callbacks if the resource is created inline or not.

what is the use case for NOT calling afterCreate/afterUpdate?

**Comment by crynobone** [comment]
> what is the use case for NOT calling afterCreate/afterUpdate?

<img width="826" alt="image" src="https://user-images.githubusercontent.com/172966/223107193-dd8dc1bb-5225-44eb-a4af-83710d87cb31.png">

The documentation can be interpreted "Resource Action" as the current resource and the current behavior for Resource Hook isn't cascaded on an inline relationship at the moment. 

Also, you can also use Model Observer and also use `Laravel\Nova\Observable` to register model events specifically for this purpose. https://nova.laravel.com/docs/4.0/resources/?#resource-events

**Reply by puzzledmonkey** [reply]
Hmm I'd say the documentation isn't specific there (it should mention inline resources being exempt) and without explicit documentation my feeling is this violates the principle of least surprise.

If there's a use case for this behavior perhaps it could be configurable (eg public static $ignoreInlineLifecycleEvents = true; or similar), or just update the documentation to make it clear.

Point taken about the Observers though, thank you. I guess it would need pairing with Nova::whenServing to serve the same purpose as afterCreate etc.

**Reply by crynobone** [reply]
As stated in the documentation, `Laravel\Nova\Observable` only register the observer for Nova requests, so you don't need to do `Nova::whenServing()`.
